lib/opal/rewriters/logical_operator_assignment.rb

```diff
@@ -0,0 +1,199 @@
+require 'opal/rewriters/base'
+
+module Opal
+  module Rewriters
+    class LogicalOperatorAssignment < Base
+      class AbstractHandler < self
+        def self.get(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.set(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.process(lhs, rhs, root_type)
+          s(root_type,
+            get(lhs, rhs),
+            set(lhs, rhs))
+        end
+
+        def self.new_temp
+          @@counter ||= 0
+          @@counter += 1
+          :"$or_equals_lhs_recvr_tmp_#{@@counter}"
+        end
+
+        def self.reset_tmp_counter!
+          @@counter = 0
+        end
+      end
+
+      # Takes    `lhs ||= rhs`
+      # Produces `lhs || lhs = rhs`
+      class LocalVariableHandler < AbstractHandler
+        def self.get(lhs, rhs)
+          lhs.updated(:lvar) # lhs
+        end
+
+        def self.set(lhs, rhs)
+          lhs.updated(:lvasgn, [*lhs, rhs]) # lhs = rhs
+        end
+      end
```

Comment:
I was thinking about something similar initially, but I just don't like the idea of having abstract class that can't do anything on its own. Having an auto-generated registry is a good idea, but there are only 7 classes. I'm not completely against it, I just don't see any reason to compilate this code. Ping me please if you still want to change, I'm ok with it.